### DISCUSSION TITLE
TheiaBackendModelServerClientV2.edit() does not work with CompoundCommand

### DISCUSSION THREAD

**Discussion by kubomann** [eclipse-emfcloud/emfcloud#209] [ModelServer] [answered]
Hi.
(I couldn't find a suitable category, so I'm posting to `All Other`.)

I am currently developing with Modelserver Theia integration\([eclipse-emfcloud/emfcloud-modelserver-theia](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia)\).
`TheiaModelServerClientV2.edit()` worked if I set the `SetCommand` as an argument.
However, it didn't work if I included `SetCommand` in `CompoundCommand` and set `CompoundCommand` as an argument.

`TheiaBackendModelServerClientV2.edit()` is implemented as follows:
```ts
    edit(modeluri: URI, patchOrCommand: PatchOrCommand, format?: Format): Promise<ModelUpdateResult> {
        if (ModelServerCommand.is(patchOrCommand)) {
            return super.edit(modeluri, ensureCommandPrototype(patchOrCommand));
        }
        return super.edit(modeluri, patchOrCommand, format);
    }
```
I think the reason is that `ModelServerCommand.is()` is false for `CompoundCommand`. Is this a bug?

**Comment by ndoschek** [comment, marked as answer]
Hi @kubomann,

Thank you for catching that!
I had a look and I would agree that this is a bug.
The if clause should check for both command types (e.g. like `if (ModelServerCommand.is(patchOrCommand) || CompoundCommand.is(patchOrCommand))`).

Would you mind [opening an issue in the `emfcloud-modelserver-theia` repository](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/issues/new/choose), as you found it :bug: 
We would also gladly accept a contribution to get this fixed :)

Thanks and BR,
Nina

**Reply by kubomann** [reply]
Hi @ndoschek .

Thank you for your answer.
I have opened an issue /eclipse-emfcloud/emfcloud-modelserver-theia#144 .

**Reply by ndoschek** [reply]
Great, thank you! :+1:
